### DISCUSSION TITLE
Want to know about DaisyUIKit

### DISCUSSION THREAD

**Discussion by keanesnow98** [saadeghi/daisyui#1800] [Q&A] [answered]
Hi, I accidentally visited daisyuikit website [here](https://daisyuikit.com).
I found it's much similar to DaisyUI original [website](https://diayui.com).
I would like to know what is the difference between daisyui and daisyuikit.

**Comment by saadeghi** [comment, marked as answer]
I didn't know about this project. Looks like it's not ready to use yet but based on the landing page, it's a Vue component library based on daisyUI.  
With daisyUI, you use CSS class names like `<button class="button button-primary">`. No JS required and you can use it on every framework.  
With daisyUI Kit (which is a different project) You can use Vue components like `<Button primary>` which probably applies daisyUI classes using Vue props.  

In my opinion, the pros of using framework specific JS components is that you don't need to use CSS class names. You use component props instead (`<Button primary>` instead of `<button class="button button-primary">`) so it's shorter...  
I admire projects like this but for any framework specific component library the downside is, you're dependent to JS even for simple HTML tags which don't need hydration. Also you would probably need to include all your JS framework components in `tailwind.config.js` and Tailwind would need to generate class names for all the props defined in the component even if you don't use them in the app. For example if you use Tailwind + daisyUI as it is with Vue, you add your Vue files to `tailwind.config.js`. If you use `btn-accent` class name, this class name will be included in your production but if you don't use it, it won't be in your production. With framework specific component libraries like daisyUI Kit you probably need to include the whole Vue component of `Button.vue` (and other components) in `tailwind.config.js` and those components include all variants of all elements so either you use or not use a variant like `btn-accent` it will be included in your production because the `Button.vue` component is probably including all the class names as props and there's no difference if you use a variant of a component as prop or not. That means larger app in production with lots of unused CSS. I don't see that efficient.

**Reply by keanesnow98** [reply]
Thanks for your explanation. I thought you're working on that DaisyUIKit project.
Maybe somebody else what to make his own framework, but it's so weird that he never contacted with you.
Anyway, thank you very much.

**Reply by saadeghi** [reply]
@keanesnow98 Well, It's just a Vue library which uses daisyUI as a dependency so It's not a violation of daisyUI MIT license. It's allowed to do so.
